gotests_test.go

```diff
@@ -492,6 +492,7 @@ func TestGenerateTests(t *testing.T) {
 			name: "Entire testdata directory",
 			args: args{
 				srcPath: `testdata/`,
+				template: "testify",
```

Comment:
What happens when you remove this? Ideally, I'd like this these existing tests to not need to change. Especially this one which was added 5 years ago, before `template` was an argument.

Reply:
I run it few times and this test generates (without pinned template)

```
 go test -v ./... -count 10 | grep TestGenerateTests/Entire_testdata_directory -A20 -B20


=== RUN   TestGenerateTests/Entire_testdata_directory
    gotests_test.go:846: "Entire testdata directory". GenerateTests(testdata/) error = output.Process: render.Header: template: no template "header" associated with template "render", wantErr false
--- FAIL: TestGenerateTests/Entire_testdata_directory (0.76s)
---------------------------------------------------------------
=== RUN   TestGenerateTests/Entire_testdata_directory
    gotests_test.go:846: "Entire testdata directory". GenerateTests(testdata/) error = output.Process: render.TestFunction: template: no template "function" associated with template "render", wantErr false
--- FAIL: TestGenerateTests/Entire_testdata_directory (0.64s)
---------------------------------------------------------------
=== RUN   TestGenerateTests/Entire_testdata_directory
--- PASS: TestGenerateTests/Entire_testdata_directory (2.49s)
---------------------------------------------------------------
=== RUN   TestGenerateTests/Entire_testdata_directory
    gotests_test.go:846: "Entire testdata directory". GenerateTests(testdata/) error = output.Process: render.TestFunction: template: no template "function" associated with template "render", wantErr false
--- FAIL: TestGenerateTests/Entire_testdata_directory (0.32s)
---------------------------------------------------------------
=== RUN   TestGenerateTests/Entire_testdata_directory
    gotests_test.go:846: "Entire testdata directory". GenerateTests(testdata/) error = output.Process: render.TestFunction: template: no template "function" associated with template "render", wantErr false
--- FAIL: TestGenerateTests/Entire_testdata_directory (0.36s)
---------------------------------------------------------------
=== RUN   TestGenerateTests/Entire_testdata_directory
    gotests_test.go:846: "Entire testdata directory". GenerateTests(testdata/) error = output.Process: render.Header: template: no template "header" associated with template "render", wantErr false
--- FAIL: TestGenerateTests/Entire_testdata_directory (0.37s)
---------------------------------------------------------------
=== RUN   TestGenerateTests/Entire_testdata_directory
    gotests_test.go:846: "Entire testdata directory". GenerateTests(testdata/) error = output.Process: render.Header: template: no template "header" associated with template "render", wantErr false
--- FAIL: TestGenerateTests/Entire_testdata_directory (0.72s)
---------------------------------------------------------------
=== RUN   TestGenerateTests/Entire_testdata_directory
    gotests_test.go:846: "Entire testdata directory". GenerateTests(testdata/) error = output.Process: render.Header: template: no template "header" associated with template "render", wantErr false
--- FAIL: TestGenerateTests/Entire_testdata_directory (0.98s)
---------------------------------------------------------------
=== RUN   TestGenerateTests/Entire_testdata_directory
    gotests_test.go:846: "Entire testdata directory". GenerateTests(testdata/) error = output.Process: render.Header: template: no template "header" associated with template "render", wantErr false
--- FAIL: TestGenerateTests/Entire_testdata_directory (0.69s)
---------------------------------------------------------------
=== RUN   TestGenerateTests/Entire_testdata_directory
    gotests_test.go:846: "Entire testdata directory". GenerateTests(testdata/) error = output.Process: render.TestFunction: template: render:65:15: executing "function" at <{{template "results" $f}}>: template "results" not defined, wantErr false
--- FAIL: TestGenerateTests/Entire_testdata_directory (1.08s)
```

Reply:
Hm... Did a stress test (`-count=10` ), it still fails sometimes at the same test

```
=== RUN   TestGenerateTests/Entire_testdata_directory
fatal error: concurrent map read and map write
```

So we have race under the stress of generateTests, should we address this now or add to the backlog?

Reply:
@butuzov Sorry for the delay. Ok I see. Let's add this race condition to the backlog.